### PR TITLE
Support a `danger local` command

### PR DESCRIPTION
**Pull request by orta**
Re: https://github.com/KrauseFx/danger/issues/17

So far I've added another commands `danger local` and set it up to use a non-existent  CI Source, the source's job is going to be to find the last PR it can find in the current history of the current repo, then fill in it's environment variables with that stuff.

### CODE REVIEW

**Comment by orta** [comment]
Now grabs enough git data :tada:

File: lib/danger/ci_source/local_git_repo.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        logs = git.log.since('2 weeks ago')
+        # Look for something like
+        # "Merge pull request #38 from KrauseFx/funky_circles\n\nAdd support for GitHub compare URLs that don't conform
+        pr_merge = logs.detect { |log| (/Merge pull request #[0-9]* from/ =~ log.message) == 0 }
+        if pr_merge
+          # then pull out the 38, to_i
+          self.pull_request_id = pr_merge.replace("Merge pull request #", "").to_i
+          self.base_commit = pr_merge.parents[0].sha
+          self.head_commit = pr_merge.parents[1].sha
+        end
```
Comment (KrauseFx) [inline lib/danger/ci_source/local_git_repo.rb]: What happens on `else`?

**Comment by KrauseFx** [comment]
Looks good, however HoundCI doesn't seem so happy

File: lib/danger/ci_source/local_git_repo.rb (hunk 1 above)
Comment (orta) [inline lib/danger/ci_source/local_git_repo.rb]: I'll do a full pass on error reporting once I've got a working concept :+1:

**Comment by orta** [comment]
also, if you have a better idea for the name, I'd be interested

File: lib/danger/ci_source/local_git_repo.rb (hunk 1 above)
Comment (KrauseFx) [inline lib/danger/ci_source/local_git_repo.rb]: :+1:

File: lib/danger/dangerfile.rb [hunk 2]
```diff
@@ -21,9 +21,32 @@ def to_s
...
+      puts "-" * (width + 4)
+
+      puts "Metadata:"
+      puts "#{'SCM'.blue}      : #{env.scm.class}"
+      puts "#{'Source'.blue}   : #{env.ci_source.class}"
+      puts "           #{'Base commit'.blue} : #{env.ci_source.base_commit}"
+      puts "           #{'HEAD commit'.blue} : #{env.ci_source.head_commit}"
+      puts "           git diff  #{env.ci_source.base_commit} #{env.ci_source.head_commit}".yellow
+      puts "#{'Requests'.blue} : #{env.request_source.class}"
+    end
```
Comment (orta) [inline lib/danger/dangerfile.rb]: this should be replaced with cork, really

Comment (orta) [inline lib/danger/dangerfile.rb]: It's a start
screen shot 2016-02-08 at 10 43 56

File: lib/danger/commands/local.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    def validate!
+      super
+      unless @dangerfile_path
+        help! "Could not find a Dangerfile."
+      end
+    end
+
+    def run
+      ENV["DANGER_USE_LOCAL_GIT"] = "YES"
```
Comment (orta) [inline lib/danger/commands/local.rb]: all `ci_source`s use ENV for validation, I think using it here makes sense.

File: lib/danger.rb [hunk 4]
```diff
@@ -1,8 +1,7 @@
 require "danger/version"
 require "danger/dangerfile"
 require "danger/environment_manager"
-require "danger/runner"
-require "danger/init"
+require "danger/commands/runner"
```
Comment (orta) [inline lib/danger.rb]: tried to make this work like cocoapods, but I'm missing something

Comment (orta) [inline lib/danger.rb]: CocoaPods shows the sub-commands
```
~/d/r/l/danger (local_test) ⏛  pod --help
Usage:

    $ pod COMMAND

      CocoaPods, the Cocoa library package manager.

Commands:

    + cache          Manipulate the CocoaPods cache
    + deintegrate    Deintegrate CocoaPods from your project.
    + dependencies   Show project's dependency graph.
```

Danger doesn't
```
~/d/r/l/danger (local_test) ⏛  bundle exec danger --help
Usage:

    $ danger COMMAND

      Run the Dangerfile.

Options:

    --version   Show the version of the tool
    --verbose   Show more debugging information
```

File: lib/danger/commands/runner.rb [hunk 5]
```diff
@@ -18,6 +21,7 @@ def validate!
     def run
       # The order of the following commands is *really* important
       dm = Dangerfile.new
+      dm.verbose = verbose
```
Comment (orta) [inline lib/danger/commands/runner.rb]: every claide object has a verbose

**Comment by orta** [comment]
OK, I've removed match, and fui - it wasn't being used. 

I've also updated Danger to this PR https://github.com/KrauseFx/danger/pull/40/ - which makes `--verbose` work - should make debugging DM easier for me.

File: lib/danger/dangerfile.rb (hunk 2 above)
Comment (segiddins) [inline lib/danger/dangerfile.rb]: terminal-table may be what you want here

Comment (KrauseFx) [inline lib/danger/dangerfile.rb]: Yeah, so simple to use and nice looking https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/print_table.rb#L19-L25

Comment (orta) [inline lib/danger/dangerfile.rb]: Cool, I was wondering if Cork would do this? Can get another external user in this case

Comment (orta) [inline lib/danger/dangerfile.rb]: but yeah, terminal-table looks great

**Comment by orta** [comment]
Punted the tables stuff to https://github.com/danger/danger/issues/62

**Pull request merged**